src/Route.php

```diff
@@ -23,19 +23,20 @@ class Route implements IRouter
 	 * @var array|null
 	 */
 	private $supportedHttpMethods;
+	
+	/**
+	 * @var PathMatcher
+	 */
+	private $pathMatcher;
 
 
-	public function __construct(string $route, string $presenterClassName , array $supportedHttpMethods = null)
+	public function __construct(string $route, string $presenterClassName ,
+			array $supportedHttpMethods = null, PathMatcher $pathMatcher = null)
```

Comment:
This is definitely not good practice and idea. Class `PathMatcher` should be replaced with interface - because you explicitly are asking for `PatchMatcher` class in constructor, but no one can create this class, because it is already existing, so it is kind of useless.

It would be much better to create interface, typehint it here and make `PatchMatcher` implementing that interface.